appendix/single-sign-on.rst

```diff
@@ -0,0 +1,521 @@
+
+.. warning:: This will take your Zammad instance **offline**
+   until Apache is fully configured and running.
+
+   If you wish to minimize downtime, you can save this step for last;
+   just bear in mind that Apache will not start
+   if the port it wants to listen on is being used by NGINX.
+
+   If for any reason you can’t complete this tutorial,
+   simply turn off Apache and restore NGINX:
+
+   .. code-block:: sh
+
+      $ systemctl stop apache2
+      $ systemctl disable apache2
+      $ systemctl enable nginx
+      $ systemctl start nginx
+
+2b. Install Apache
+^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   # Ubuntu & Debian
+   $ apt update
+   $ apt install apache2 krb5-user libapache2-mod-auth-kerb
+
+   # CentOS
+   $ yum install httpd krb5-workstation mod_auth_kerb
+
+   # openSUSE
+   $ zypper ref
+   $ zypper install apache2 krb5-client apache2-mod_auth_kerb
+
+2c. Enable Apache modules
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   # Ubuntu, Debian, & openSUSE
+   $ a2enmod auth_kerb headers rewrite proxy proxy_html proxy_http proxy_wstunnel
+
+On systems without ``a2enmod`` (*e.g.,* CentOS),
+add/uncomment the appropriate ``LoadModule`` statements
+in your Apache config:
+
+.. code-block::
+
+   # /etc/httpd/conf/httpd.conf
+
+   LoadModule auth_kerb_module /usr/lib/apache2/modules/mod_auth_kerb.so
+   LoadModule headers_module modules/mod_headers.so
+   LoadModule rewrite_module modules/mod_rewrite.so
+   LoadModule proxy_module modules/mod_proxy.so
+   LoadModule proxy_html_module modules/mod_proxy_html.so
+   LoadModule proxy_http_module modules/mod_proxy_http.so
+   LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so
+
+2d. Configure Kerberos
+^^^^^^^^^^^^^^^^^^^^^^
+
+Kerberos realm configuration is how you tell the Zammad server
+about the *domain controller* (Active Directory server).
+
+.. note:: Replace the following placeholders in the sample config below:
+
+   :``<domain>``:                   Windows domain
+   :``<domain-controller>``:        Domain controller IP/FQDN(s)
+   :``<master-domain-controller>``: Master domain controller IP/FQDN
+
+                                    (must not be read-only,
+                                    but can be the same as ``<domain-controller>``)
+
+.. code-block::
+
+   # /etc/krb5.conf
+
+   [libdefaults]
+     default_realm = <domain>
+     default_tkt_enctypes = aes256-cts-hmac-sha1-96
+     default_tgs_enctypes = aes256-cts-hmac-sha1-96
+     permitted_enctypes = aes256-cts-hmac-sha1-96
+
+     kdc_timesync = 1
+     ccache_type = 4
+     forwardable = true
+     proxiable = true
+     fcc-mit-ticketflags = true
+
+   [realms]
+           # multiple KDCs ok (one per line)
+           <domain> = {
+                   kdc = <domain-controller>
+                   admin_server = <master-domain-controller>
+                   default_domain = <domain>
+           }
+
+   [domain_realm]
+           .<domain> = <domain>
```

Comment:
I removed a comment here about the leading `.` not being a typo because I wanted to keep the sample configs as clean as possible, but I'm open to putting it back if you feel strongly about it.

Reply:
I felt that it may be confusing but if you say it didn't confuse you and you understood why it's there, I'm fine with leaving it as is. :)

Reply:
Thanks. I do prefer to avoid telling the reader anything whenever possible, because we're already telling them a _lot,_ and each new piece of information adds to the cognitive burden of getting through the guide. I think it also helps that the line is almost duplicated just below:

```
.<domain> = <domain>
<domain> = <domain>
```

which would make it _very_ strange if the dot were a typo.

In any case, we'll know if we need to put it back when we do (or don't) get complaints from the readers, right? :wink: